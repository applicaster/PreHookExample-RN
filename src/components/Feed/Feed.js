import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  View,
  ScrollView,
  RefreshControl,
  Image,
  Text,
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    paddingTop: 20,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#27DA86',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
    color: '#ffffff',
  },
  subTitle: {
    textAlign: 'center',
    color: '#ffffff',
    marginBottom: 5,
    opacity: 0.85,
  },
});

class Feed extends Component {
  constructor(props) {
    super(props);
    this.onRefresh = this.onRefresh.bind(this);
  }

  componentWillMount() {
    this.props.setAccountId('595de7ac8a7c43000914c5a5');
    this.props.setTimelineId('595df85282e4a41a3e37374b');
    this.props.setTimezone('3600');
    this.props.setEnvironment('production');
    this.props.fetchSocialEvents();
  }

  onRefresh() {
    this.props.fetchSocialEvents();
  }

  renderCaption(caption) {
    return (caption && caption.text) ? caption.text : null;
  }

  render() {
    const socialPosts = this.props.socialPosts;
    const refreshControl = (
      <RefreshControl
        refreshing={this.props.loading}
        onRefresh={this.onRefresh}
      />);

    return (
      <View style={styles.container}>
        <ScrollView refreshControl={refreshControl}>
          {socialPosts.map(post =>
            <View key={post.id}>
              <Image
                style={{ width: 360, height: 360 }}
                source={{ uri: post.images.low_resolution.url }}
              />
              <Text>{this.renderCaption(post.caption)}</Text>
            </View>
          )}
        </ScrollView>
        <Text style={styles.subTitle}>
          by Applicaster
        </Text>
      </View>
    );
  }
}

Feed.propTypes = {
  loading: PropTypes.bool,
  socialPosts: PropTypes.array,
  fetchSocialEvents: PropTypes.func,
  setAccountId: PropTypes.func,
  setTimelineId: PropTypes.func,
  setTimezone: PropTypes.func,
  setEnvironment: PropTypes.func,
};

export default Feed;
