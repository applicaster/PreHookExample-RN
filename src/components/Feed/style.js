import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  feed: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#323232',
  },
  headerStyle: {
    backgroundColor: '#323232',
    borderWidth: StyleSheet.hairlineWidth,
    borderBottomColor: 'rgba(255,255,255,0.3)',
  },
  headerTitleStyle: {
    color: '#7ED321',
  },
});
