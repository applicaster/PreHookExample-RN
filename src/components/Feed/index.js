import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  fetchSocialEvents,
  setAccountId,
  setTimelineId,
  setTimezone,
  setEnvironment,
} from '../../actions';
import Feed from './Feed';

const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchSocialEvents,
  setAccountId,
  setTimelineId,
  setTimezone,
  setEnvironment,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Feed);
