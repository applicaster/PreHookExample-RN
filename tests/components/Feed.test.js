import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Feed from '../../src/components/Feed/Feed';

const event = (id) => ({
  id,
  user: {
    full_name: 'Carlos Pinto',
    username: 'cpinto7',
    profile_picture: 'url',
  },
  images: {
    standard_resolution: {
      url: 'url',
    },
    caption: 'caption',
  },
});

describe('Feed', () => {
  test('component renders correctly without events', () => {
    const tree = renderer.create(
      <Feed
        fetchSocialEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading={false}
        socialEvents={[]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly with events', () => {
    const tree = renderer.create(
      <Feed
        fetchSocialEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading={false}
        socialEvents={[event(1), event(2)]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly while loading', () => {
    const tree = renderer.create(
      <Feed
        fetchSocialEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading={true}
        socialEvents={[]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly while loading with events', () => {
    const tree = renderer.create(
      <Feed
        fetchSocialEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading={true}
        socialEvents={[event(1), event(2)]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
