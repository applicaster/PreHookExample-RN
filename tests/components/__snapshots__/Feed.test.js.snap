// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed component renders correctly while loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#27DA86",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
      />
    }
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;

exports[`Feed component renders correctly while loading with events 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#27DA86",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "url",
            }
          }
          style={
            Object {
              "height": 750,
              "width": 750,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
      </View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "url",
            }
          }
          style={
            Object {
              "height": 750,
              "width": 750,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Feed component renders correctly with events 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#27DA86",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "url",
            }
          }
          style={
            Object {
              "height": 750,
              "width": 750,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
      </View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "url",
            }
          }
          style={
            Object {
              "height": 750,
              "width": 750,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Feed component renders correctly without events 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#27DA86",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
